Assets/Scripts/Minigames/MinigameBarSetup.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+class MinigameInstantiateLater
+{
+    public Sprite sprite;
+    public string name;
+
+    public MinigameInstantiateLater(Sprite s, string n)
+    {
+        sprite = s;
+        name = n;
+    }
+}
+
+[ExecuteAlways]
+public class MinigameBarSetup : MonoBehaviour
+{
+    public Transform staminaFillTransform;
+
+    [Header("Preset Assets")]
+    public GameObject segmentBasePrefab;
+
+    public Sprite startSegmentSprite;
+    private readonly string startSegmentName = "StartSegment";
+    public Sprite middleSegmentSprite;
+    private readonly string middleSegmentName = "MiddleSegment";
+    public Sprite endSegmentSprite;
+    private readonly string endSegmentName = "EndSegment";
+
+    public enum MinigameStaminaType
+    {
+        Time,
+        Target
+    }
+    public MinigameStaminaType staminaType;
+    public Sprite segmentSprite;
+    public List<Image> childImages;
+
+    private void OnValidate()
+    {
+        UpdateListOfChildren();
+    }
+
+    void UpdateListOfChildren()
+    {
+        if (staminaType != MinigameStaminaType.Target)
+        {
+            childImages.Clear();
+            foreach (Transform child in staminaFillTransform)
+            {
+                // because we cannot destroy objects in OnValidate, we wait until the end of the frame for this:
+                // https://answers.unity.com/questions/1318576/destroy-child-objects-onvalidate.html
+                StartCoroutine(DestroyAtEndOfFrame(child.gameObject));
+            }
+        }
+        else
```

Comment:
Any ideas on cleaning this code would be helpful. It's basically attempting to cover all cases when people can delete or add children to this stamina bar transform.